src/Hedgehog.Experimental/Gen.fs

```diff
@@ -525,33 +562,41 @@ module GenX =
               return values.GetValue index |> unbox
             }
 
+        | Shape.Collection s ->
+            s.Accept {
+              new ICollectionVisitor<Gen<'a>> with
+              member _.Visit<'collection, 'element when 'collection :> System.Collections.Generic.ICollection<'element>> () =
+                match typeShape with
+                | Shape.Poco (:? ShapePoco<'a> as shape) ->
+                  gen {
+                    let! collectionCtor = genPoco shape
+                    let! count =
+                      if canRecurse typeof<'element> then
+                        Gen.integral config.SeqRange
+                      else
+                        Gen.constant 0
+                    let! elements =
+                      incrementRecursionDepth typeof<'element>
+                      |> autoInner config
+                      |> List.replicate count
+                      |> ListGen.sequence
```

Comment:
Now that I think about it, we shouldn't have to explicitly generate `count` and `elements`.  Instead, we can generate an F# list (which first generates a `count` and then the `elements`) and then copy those elements into the collection.

Reply:
This PR started out as mostly a copy/paste of Shape.FSharpList. Looks like we got back there in the end :)